src/reject.rs

```diff
@@ -107,14 +116,47 @@ impl Rejection {
     pub fn status(&self) -> http::StatusCode {
         Reject::status(self)
     }
+
+    /// Add given `err` into `Rejection`.
+    pub fn with<E>(self, err: E) -> Self
+    where E: Into<Cause> + Sized
+    {
+        let cause = Some(err.into());
+        Self {
+            cause,
+            .. self
+        }
+    }
+}
+
+impl PartialOrd for Rejection {
```

Comment:
We might not want ordering to be part of the public API of a `Rejection`... Mostly because I don't know if it's obvious to a user *why* it has one. What would they assume the ordering means? The ordering used on the `reason` field can just be an implementation deal in `CombineRejection`.